datanode/dehistory/fsutil/fsutil.go

```diff
@@ -0,0 +1,200 @@
+package fsutil
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	vgfs "code.vegaprotocol.io/vega/libs/fs"
+)
+
+var ErrMismatchedDeviceVersion = errors.New("mismatched device versions")
+
+// RemoveAllFromDirectory used in place of os.RemoveAll when the directory should be emptied but not removed.
+func RemoveAllFromDirectory(dir string) error {
+	exists, err := vgfs.PathExists(dir)
+	if err != nil {
+		return err
+	}
+	if !exists {
+		return nil
```

Comment:
super minor but maybe it should be an error if the dir doesn't exist?

Reply:
ah yeah, thats nto clear, renamed the method to RemoveAllFromDirectoryIfExists